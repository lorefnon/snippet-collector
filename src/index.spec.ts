import { collectSnippetsFromSource, collectSnippetsFromFile, consolidateSnippets } from './index';

test('collectSnippetsFromFile', async () => {
    const snippets = new Map();
    await collectSnippetsFromFile('src/__fixtures__/test.js', snippets);
    expect(snippets).toMatchSnapshot();
    const consolidated = consolidateSnippets(snippets);
    expect(consolidated).toMatchSnapshot();
})