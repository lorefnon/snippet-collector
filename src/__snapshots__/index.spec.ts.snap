// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectSnippetsFromFile 1`] = `
Map {
  "usages" => Array [
    Object {
      "content": "import minimist from \\"minimist\\";
",
      "endLineNo": 3,
      "name": "usages",
      "occuranceIndex": 0,
      "startLineNo": 1,
    },
    Object {
      "content": "const extractedOptions = minimist(process.argv.slice(1), {
    aliases: {
        'o': 'options'
    }
})
",
      "endLineNo": 14,
      "name": "usages",
      "occuranceIndex": 1,
      "startLineNo": 8,
    },
  ],
  "simple-usage" => Array [
    Object {
      "content": "import minimist from \\"minimist\\";

const options = minimist(process.argv.slice(1));
",
      "endLineNo": 6,
      "name": "simple-usage",
      "occuranceIndex": 0,
      "startLineNo": 1,
    },
  ],
}
`;

exports[`collectSnippetsFromFile 2`] = `
Object {
  "simple-usage": Object {
    "content": "import minimist from \\"minimist\\";

const options = minimist(process.argv.slice(1));
",
    "name": "simple-usage",
  },
  "usages": Object {
    "content": "import minimist from \\"minimist\\";
const extractedOptions = minimist(process.argv.slice(1), {
    aliases: {
        'o': 'options'
    }
})
",
    "name": "usages",
  },
}
`;
